src/lava/lib/optimization/apps/scheduler/solver.py

```diff
@@ -0,0 +1,289 @@
+# Copyright (C) 2023 Intel Corporation
+# SPDX-License-Identifier: BSD-3-Clause
+# See: https://spdx.org/licenses/
+
+
+import networkx as ntx
+import numpy as np
+
+from networkx.algorithms.approximation import maximum_independent_set
+from typing import List, Dict, Tuple, Optional
+
+import matplotlib.pyplot as plt
+from matplotlib.patches import PathPatch
+from matplotlib.path import Path
+
+from lava.utils import loihi
+from lava.lib.optimization.apps.scheduler.problems import \
+    (SchedulingProblem, SatelliteScheduleProblem)
+from lava.lib.optimization.apps.scheduler.utils.util_funcs import *
+from lava.lib.optimization.problems.problems import QUBO
+from lava.lib.optimization.solvers.generic.solver import (OptimizationSolver,
+                                                          SolverConfig)
+from lava.lib.optimization.utils.generators.mis import MISProblem
+
+
+class Scheduler:
+
+    def __init__(self,
+                 sp: SchedulingProblem,
+                 qubo_weights: Tuple[int, int] = (1, 8),
+                 probe_cost: bool = False,
+                 probe_loihi_exec_time=False,
+                 probe_loihi_energy=False):
+        """Solver for Scheduling Problems.
+
+        Parameters
+        ----------
+        sp (SchedulingProblem) : Scheduling problem object as defined in
```

Comment:
For the docstring parameters, could you use the same design as e.g. in problems.py?

Reply:
Done.